### PR TITLE
Add CPU architecture, SMT state and threads per core to collected data

### PR DESCRIPTION
**Pull request by sere**

### CODE REVIEW

**Comment by elfosardo** [comment]
@sere please squash the commits

**Comment by sere** [comment]
> @sere please squash the commits

Done :)

File: hardware/detect.py [hunk 1]
```diff
@@ -709,6 +709,11 @@ def _get_governor(lcpu):
             lscpux[item.strip(':')] = value.strip()
 
     hw_lst.append(("cpu", "physical", "number", int(lscpu["Socket(s)"])))
+
+    with contextlib.suppress(IOError):
+        value = _from_file("/sys/devices/system/cpu/smt/control")
```
Comment (ErwanAliasr1) [inline hardware/detect.py]: Don't you want active instead of control ?

File: hardware/detect.py [hunk 2]
```diff
@@ -726,6 +731,7 @@ def _get_governor(lcpu):
                                      ('family', 'CPU family', int),
                                      ('model', 'Model', _maybe_int),
                                      ('stepping', 'Stepping', _maybe_int),
+                                     ('architecture', 'Architecture', None),
```
Comment (ErwanAliasr1) [inline hardware/detect.py]: I don't see how this is correlated to the patch

Comment (sere) [reply hardware/detect.py]: The patch adds two fields to the output: CPU architecture field and SMT status to the collected data. They were two independent patches that I squashed as suggested by @elfosardo. I updated the title of the merge request to make it less confusing.

File: hardware/detect.py (hunk 1 above)
Comment (sere) [reply hardware/detect.py]: They have a different output, but they can both be used to read the SMT state. "active" has values 0 and 1, "control" has values "on" "off" "forceoff" "notsupported" "notimplemented".

Following some tests we found out that, in some cases, there might be a discrepancy between the value in "active" and "control", with active with value 0 or off and control with value on. This happens when SMT is enabled but the siblings are offline.

Given this discrepancy and the fact that both this measures don't extract the number of threads per core, I suggest to expose instead the field "Thread(s) per core" of lscpu. Would you agree to this change?

File: hardware/detect.py (hunk 2 above)
Comment (ErwanAliasr1) [reply hardware/detect.py]: Yeah sorry I missed that.

File: hardware/detect.py (hunk 1 above)
Comment (ErwanAliasr1) [reply hardware/detect.py]: I didn't tested this as much as it seems you did so I'll trust your testing. Let's use control then.
Yeah, reported threads per core is a good idea.

Comment (ErwanAliasr1) [reply hardware/detect.py]: It would be a good idea to have some tests for the smt value.

Comment (sere) [reply hardware/detect.py]: OK, I'll implement both threads per core and SMT using "control".

What do you have in mind as tests for the smt value?

Comment (ErwanAliasr1) [reply hardware/detect.py]: Something like the boost testing in test_get_cpus_vm(). That's very simple.

Comment (sere) [reply hardware/detect.py]: Are you talking about

`     for i in range(1):
        calls.append(mock.call('/sys/devices/system/cpu/cpufreq/boost'))`
?

This kind of test is already there for `smt` too.

**Comment by elfosardo** [comment]
@sere also needs a rebase

**Comment by sere** [comment]
I rebased and added threads_per_core. The commit message now is a bit more descriptive.

**Comment by ErwanAliasr1** [comment]
Thanks for your contribution.

**Pull request merged**